### ISSUE TITLE
wind.flutes.whistle.tin does not designate a valid sound ID

### ISSUE THREAD

**Issue by riaanvn** [w3c/musicxml#289]
There is no such instrument as Tin Whistle without a key indicator, e.g.
 sound id="wind.flutes.whistle.tin"

In musicxml/schema/sounds.xml
The most common key is Tin Whistle in D, e.g. 
sound id="wind.flutes.whistle.tin.d"

The generic entry 
sound id="wind.flutes.whistle.tin"

should either be
a) removed, or 
b) (my recommendation) modified to reflect an actual instrument, Tin Whistle in C, e.g.
sound id="wind.flutes.whistle.tin.c"
which would mean the top 3 tin whistle keys (D, C, Bb) are represented. 

As justification / support, I offer 
a) a message written on the MuseScore forum (I wrote the original post)
https://musescore.org/en/node/292977
b) 3rd party Tin Whistle instrument definitions
https://github.com/jgadsden/musescore-tin-whistle
that already make this distinction. This instrument set does have Tin Whistle in C instead 
wind.flutes.whistle.tin.c
and a keyless Tin Whistle (which is an alternate way to refer to the Tin Whistle in D, e.g. these are the same instrument
wind.flutes.whistle.tin.d 
wind.flutes.whistle.tin.common

**Comment by mdgood** [comment]
@riaanvn Thank you for raising this issue. The MusicXML sounds list generally always tries to include a more generic version of a name in the hierarchy. I think the best way to approach this would be to add a wind.flutes.whistle.tin.c value rather than renaming or removing the wind.flutes.whistle.tin value.

If I am understand the issue correctly that should meet your needs. Please let me know if I am misunderstanding something.

**Comment by riaanvn** [comment]
Hi there @mdgood 

Thank you for your response. 

Adding wind.flutes.whistle.tin.c is a step in the right direction. 

Unfortunately leaving a generic version of a tin whistle will not clear up the (very real, not just potential) confusion. Since the tin whistle does not have a default key, having a generic version of the instrument is bad for the following reasons:

a) The Tin Whistle in D is considered the most common key for the instrument, as per
https://en.wikipedia.org/wiki/Tin_whistle#Whistle_keys
(note the wikipedia article is somewhat ambiguous - a Tin Whistle in D plays both in D and G major, they are not referring to two different models, in D and in G)
but is by no means so common that it could be considered the default. Within in a musical context, you would never reference the tin whistle (e.g. in parts or when for sale) without the key. 

b) To contrast with at another, very standardised instrument: brass.trumpet . The trumpet in Bb is the default, and does not need the key, e.g. brass.trumpet.bb. The tin whistle does not have such a default key, meaning  
wind.flutes.whistle.tin
could mean one of the following:

i) Tin Whistle in C, because the C is implied in many western instruments, and because wind.flutes.whistle.tin.c is the missing of the three most common variants.  
ii) Tin Whistle in D, because that is the most common key for this instrument instrument. 
Users of this file entry have absolutely no way of knowing which of the two. 

I worked on the assumption of (i) being assumen, which is why @jgadsden and I submitted a PR for that. 

Apologies for the long-winded explanation. Hope I was able to make a convincing case.

**Comment by mdgood** [comment]
Thanks for the additional details, @riaanvn. In MusicXML sounds, I would suggest that something like wind.flutes.whistle.tin instead means:

iii) Whatever Tin Whistle sound you have available as an application-specific default.

I definitely see the confusion without having a wind.flutes.whistle.tin.c definition. In your trumpet example though, MusicXML does indeed have a brass.trumpet.bflat sound in addition to brass.trumpet. All sounds of the form type.family.instrument.variation also have a type.family.instrument sound included. The same is true for sounds with 2 dots or 4 dots in the name, in addition to those with 3 dots in the name.

The ability to selectively encode how much detail is available in MusicXML data is a design principle that runs throughout the format. MNX-Common will likely scale this back to better support the new MNX use cases. For MusicXML 3.2, I believe that we want to maintain consistency with both prior versions and this overall design style.

This is why I would like wind.flutes.whistle.tin to stay, while also adding a wind.flutes.whistle.tin.c sound. I think this will address the confusion while keeping the design of the entire set of sound names consistent.

**Comment by mdgood** [comment]
@riaanvn and @jgadsden, could you please both join the W3C Music Notation Community Group? Membership is free of charge and will allow us to accept a pull request. If you go to our home page at https://www.w3.org/community/music-notation/ and click the JOIN OR LEAVE THIS GROUP button, that will guide you through the process of setting up a W3C account and then joining this community group. Thanks!

**Comment by riaanvn** [comment]
Thank you @mgood. I joined. 
I have also created a new PR, https://github.com/w3c/musicxml/pull/299 , to add whistle.tin.c . 

@jgadsden, you may close https://github.com/w3c/musicxml/pull/290

**Comment by riaanvn** [comment]
@mgood - for my own benefit 
@gadsden and I are fairly active in the MuseScore community, and need to understand what the implication is of not making the key explicit in the instrument name and the following two scenarios:
(a) renaming whistle.tin to whistle.tin.c vs
(b) adding whistle.tin.c as a separate definition of the instrument (whistle.tin is currently configure as an instrument in C, e.g. an implied C, which we want to change to make it explicit)

Is the instrument definition included in an exported .mxl. e.g. (using real-world product, but not necessary their default pitch for the instrument)?
If a score from MuseScore using whistle.tin (with a pitch definition of C) is exported to MusicXML, then imported in Finale, which has a pitch definition of D for whistle.tin, it will be 
i) imported as in C (e.g. what the source export has defined) or 
I) adopt pitch definition of D?

**Comment by mgood** [comment]
@riaanvn  you mean @mdgood

**Comment by mdgood** [comment]
@riaanvn, these instrument definitions only refer to instrument sounds. They only affect playback, not any notation elements such as key signature or transposition. The whistle.tin sound could refer to the sound of either a C, D, or Bb whistle, whichever one an application uses as a default.

**Comment by riaanvn** [comment]
Thanks @mdgood

Let me rephrase the above example with a focus on playback: If two different programs have different keys for the defaults instrument (one C, one D), will it play back (or a chance of) if a score that was exported from the program with a default instrument of C, opened up in another program where the default is D?

**Comment by riaanvn** [comment]
Hi @mdgood 
I have a PR open on MuseScore to have whistle.tin renamed to whistle.tin.c
https://github.com/musescore/MuseScore/pull/5556
which I will be able to get merged as soon as you are able to merge 
https://github.com/w3c/musicxml/pull/299

**Comment by mdgood** [comment]
@riaanvn Thank you for the pull request. Could you please join the W3C Music Notation Community Group? Membership is free of charge, and we need the intellectual property commitment before accepting a pull request as a contribution.

If you go to our [Community Group home page](https://www.w3.org/community/music-notation/ ) and click on JOIN OR LEAVE THIS GROUP, that will guide you through the process of setting up a W3C account and joining this community group.

**Comment by mdgood** [comment]
Thanks @riaanvn for joining the Community Group! I have marked this issue In Progress. Since we have had quite a bit of discussion on this already, I will plan to merge the pull request early next week unless there are objections.
